src/Form/WidgetsConfig.php

```diff
@@ -0,0 +1,225 @@
+        '#title' => $this->t('Weight for @widget widget', ['@widget' => $widget->label()]),
+        '#title_display' => 'invisible',
+        '#attributes' => [
+          'class' => ['variant-weight'],
+        ],
+      ];
+      $form['widgets']['table'][$uuid] = $row;
+    }
+    return $form;
+  }
+
+  /**
+   * @param $form
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   */
+  public static function submitAddWidget($form, FormStateInterface $form_state) {
+    $cached_values = $form_state->getTemporaryValue('wizard');
+    /** @var \Drupal\entity_browser\EntityBrowserInterface $entity_browser */
+    $entity_browser = $cached_values['entity_browser'];
+    $widgets_num = count($entity_browser->getWidgets());
+    $widget = $form_state->getValue('widget');
+    $weight = $widgets_num + 1;
+    $entity_browser->addWidget([
+      'id' => $widget,
+      'label' => $widget,
+      'weight' => $weight,
+      // Configuration will be set on the widgets page.
+      'settings' => [],
+    ]);
+    \Drupal::service('user.shared_tempstore')
+      ->get('entity_browser.config')
+      ->set($entity_browser->id(), $cached_values);
+    $form_state->setRebuild();
+  }
+
+  /**
+   * @param $form
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   */
+  public static function submitDeleteWidget($form, FormStateInterface $form_state) {
+    $cached_values = $form_state->getTemporaryValue('wizard');
+    /** @var \Drupal\entity_browser\EntityBrowserInterface $entity_browser */
+    $entity_browser = $cached_values['entity_browser'];
+    $entity_browser->deleteWidget($entity_browser->getWidget($form_state->getTriggeringElement()['#arguments']));
+    \Drupal::service('user.shared_tempstore')
+      ->get('entity_browser.config')
+      ->set($entity_browser->id(), $cached_values);
+    $form_state->setRebuild();
+  }
+
+  /**
+   * @param $form
+   * @param $form_state
+   * @return mixed
+   */
+  public static function addWidgetCallback($form, $form_state) {
+    return $form['widgets'];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validateForm(array &$form, FormStateInterface $form_state) {
+    /** @var \Drupal\entity_browser\EntityBrowserInterface $entity_browser */
+    $entity_browser = $form_state->getTemporaryValue('wizard')['entity_browser'];
+    /** @var \Drupal\entity_browser\WidgetInterface $widget */
+    foreach ($entity_browser->getWidgets() as $widget) {
+      $widget->validateConfigurationForm($form, $form_state);
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    /** @var \Drupal\entity_browser\EntityBrowserInterface $entity_browser */
+    $entity_browser = $form_state->getTemporaryValue('wizard')['entity_browser'];
+    /** @var \Drupal\entity_browser\WidgetInterface $widget */
+    foreach ($entity_browser->getWidgets() as $uuid => $widget) {
+      $widget->submitConfigurationForm($form, $form_state);
```

Comment:
Weights are now saved correctly, but we're still missing other configuration (label, widget's own config - upload location on upload widget for example).